agents/pfeiffer_tpg366/pfeiffer_tpg366_agent.py

```diff
@@ -0,0 +1,179 @@
+        self.active = True  
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.f_sample = f_sample
+        self.take_data = False
+        self.module: Optional[Module] = None
+        self.gauge = pfeiffer()
+        self.port = port
+        self.ip_address = ip_address
+       
+        agg_params = {'frame length':60,}
+
+        self.agent.register_feed('pressures',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+    
+    def start_acq(self,session, params = None):
+        """
+        Get pressures from the Pfeiffer gauges, publishes them to the feed
+        
+        Args: 
+            sampling_frequency- defaults to 2.5 Hz
+
+        """
+        if params is None:
+            params = {}
+
+        f_sample = params.get('sampling_frequency')
+        if f_sample is None:
+            f_sample = self.f_sample
+
+        sleep_time = 1./f_sample - 0.01
+
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+        # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn("Could not start init because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'pressures',
+                    'data': {}
+                }
+                pressure_array = self.gauge.read_pressure_all()
+                #Loop through all the channels on the device
+                for i in range(len(pressure_array)):
+                    data['data']["pressure_ch"+str(i)]= pressure_array[i]
+                
+                self.agent.publish_to_feed('pressures',data)
+                time.sleep(sleep_time)
+            
+            self.agents.feeds['pressures'].flush_buffer()
+        return True, 'Acquistion exited cleanly'
+
+    
+    def stop_acq(self,session, params = None):
+        """
+        End pressure data acquisition
+        """
+        if self.take_data:
+            self.take_data = False
+            self.gauge.close()
+            return True, 'requested to stop taking data.'
+        else:
+            return False, 'acq is not currently running'
+
+
+
+if __name__ == '__main__':
+        parser = site_config.add_arguments()
+
+        pgroup = parser.add_argument_group('Agent Options')
+        pgroup.add_argument('--ip_address')
+        pgroup.add_argument('--port')
+
+        args = parser.parse_args() 
+
+        site_config.reparse_args(args, 'pfeifferAgent')
+        agent, runner = ocs_agent.init_site_agent(args)
+        pfeiffer_agent = pfeifferAgent(agent) 
```

Comment:
You grab the ip_address and port in argparse, pass them to the pfeifferAgent object here. This will allow users to configure the address and port in their ocs config file, and avoid the hardcoded address issue.